### PR TITLE
Renaming

### PR DESCRIPTION
**Pull request by riccardocadei**
Update website examples and variable names
Rename `t_anom` variable in `t_ext`
Rename `effect_modifier` variable in `intervention_vars`
Rename `lasso_rules_filter()` in `discover_causal_rules()`
Rename `split_data()` in `honest_splitting()`
Rename `prune_rules()` in ``filter_irrelevant_rules()`
Rename `discard_correlated_rules()` in `filter_correlated_rules()`
Rename `discard_anomalous_rules()` in `filter_extreme_rules()`

### CODE REVIEW

File: R/cre.R [hunk 1]
```diff
@@ -109,6 +108,7 @@
 cre <- function(y, z, X, method_params, hyper_params){
 
   # Input checks ---------------------------------------------------------------
+  logger::log_info("Load dataset")
```
Comment (Naeemkh) [inline R/cre.R:111]: If this is the beginning of the process, let's use present participle when the process starts with trailing ... 
For example, this should be: Loading dataset ...

File: R/cre.R [hunk 2]
```diff
@@ -118,13 +118,13 @@ cre <- function(y, z, X, method_params, hyper_params){
   list2env(method_params, envir = environment())
   list2env(hyper_params, envir = environment())
 
-  # Split data -----------------------------------------------------------------
-  logger::log_info("Working on splitting data ... ")
+  # Honest Splitting -----------------------------------------------------------
+  logger::log_info("Honest Splitting")
```
Comment (Naeemkh) [inline R/cre.R:122]: Same here. When reading the log file it is hard to understand what is happening.

File: R/cre.R [hunk 3]
```diff
@@ -141,10 +141,10 @@ cre <- function(y, z, X, method_params, hyper_params){
 
   # Discovery ------------------------------------------------------------------
 
-  logger::log_info("Conducting Discovery Subsample Analysis ... ")
+  logger::log_info("1. Causal Rules Discovery")
```
Comment (Naeemkh) [inline R/cre.R:144]: Same here.  For example, Working on Causal Rules Discovery ...

File: R/cre.R [hunk 4]
```diff
@@ -141,10 +141,10 @@ cre <- function(y, z, X, method_params, hyper_params){
...
 
   # Estimate ITE -----------------------
-  logger::log_info("Estimating ITE ... ")
+  logger::log_info("1.1 Estimating ITE")
```
Comment (Naeemkh) [inline R/cre.R:147]: Same here.

File: R/cre.R [hunk 5]
```diff
@@ -167,17 +167,16 @@ cre <- function(y, z, X, method_params, hyper_params){
...
-  logger::log_info("{M_final} significant Causal Rules were discovered.")
+
 
 
   # Inference ------------------------------------------------------------------
 
-  logger::log_info("Conducting Inference Subsample Analysis ...")
-  message("Conducting Inference Subsample Analysis")
+  logger::log_info("2. CATE Inference")
+  #message("Conducting Inference Subsample Analysis")
```
Comment (Naeemkh) [inline R/cre.R:176]: Also this one.

File: R/cre.R [hunk 6]
```diff
@@ -167,17 +167,16 @@ cre <- function(y, z, X, method_params, hyper_params){
...
 
-  # Estimate ITE -----------------------
-  logger::log_info("Estimating ITE ... ")
+  # Estimate ITE ---------------------------------------------------------------
+  logger::log_info("2.1 Estimating ITE")
```
Comment (Naeemkh) [inline R/cre.R:179]: Also this one. Please fix all of them.

File: R/discover_causal_rules.R [hunk 7]
```diff
@@ -15,19 +15,20 @@
...
                                 stability_selection, cutoff, pfer) {
 
   `%>%` <- magrittr::`%>%`
   rules <- NULL
   if (length(rules_list)>1){
 
     if (stability_selection) {
-      # LASSO Stability Selection
+      # Stability Selection
+      # TO DO: replace LASSO with randomization-based tests
```
Comment (Naeemkh) [inline R/discover_causal_rules.R:31]: Please use TODO instead of TO DO.

File: R/discover_causal_rules.R [hunk 8]
```diff
@@ -37,7 +38,8 @@ lasso_rules_filter <- function(rules_matrix_std, rules_list, ite_std,
       select_rules <- rule_stab
 
     } else {
-      # (Vanilla) LASSO
+      # vanilla
+      # TO DO: replace LASSO with randomization-based tests
```
Comment (Naeemkh) [inline R/discover_causal_rules.R:42]: Same here. TODO

File: R/filter_irrelevant_rules.R [hunk 9]
```diff
@@ -1,9 +1,10 @@
 #' @title
-#' Prune (Causal) Decision Rules
+#' Filter Irrelevant Decision Rules
```
Comment (Naeemkh) [inline R/filter_irrelevant_rules.R:2]: Why it is irrelevant? Is it a word that is used in the paper or literature. Pruning does make more sense in this context.

File: R/generate_causal_rules.R [hunk 10]
```diff
@@ -16,14 +16,11 @@
...
 generate_causal_rules <- function(X, ite_std, method_params, hyper_params) {
 
-  # Filter only Effect Modifiers -------
-  effect_modifiers = getElement(hyper_params,"effect_modifiers")
-  if (!is.null(effect_modifiers)) X <- X[,effect_modifiers,drop=FALSE]
-
-  # 1. Generate rules --------------------
-  logger::log_info("Rules generation ... ")
+  # Generate rules -------------------------------------------------------------
+  logger::log_info("1.2 Rules generation")
```
Comment (Naeemkh) [inline R/generate_causal_rules.R:20]: Same problem with starting a process log.

**Review by Naeemkh — CHANGES_REQUESTED** [review]
Thanks, @riccardocadei. 
Please see the comments on the code. 
Also, test-filter_correlated_rules tests raises two warnings in line 57 and 61.  Please resolve them. 
The warning message is: `the standard deviation is zero`
Please fix them and resubmit.

**Comment by riccardocadei** [comment]
@Naeemkh do you think is generally better removing the steps counting in the logging?

**Comment by Naeemkh** [comment]
> @Naeemkh do you think is generally better removing the steps counting in the logging?

Yeah. Let's drop the numbers.

File: R/filter_irrelevant_rules.R (hunk 9 above)
Comment (riccardocadei) [reply R/filter_irrelevant_rules.R:2]: There are several possible criteria to prune a tree. Here in particular we are pruning the irrelevant rules defined as the rules with no significant improvement for prediction (see 'Interpreting tree ensembles with inTrees' by Houtao Deng, 2019). I have already added this reference in the function description. 
I called the function filter_irrelevant also for consistency with the following filtering criteria

**Comment by riccardocadei** [comment]
> Thanks, @riccardocadei. Please see the comments on the code. Also, test-filter_correlated_rules tests raises two warnings in line 57 and 61. Please resolve them. The warning message is: `the standard deviation is zero` Please fix them and resubmit.

Warning suppressed

**Comment by Naeemkh** [comment]
@riccardocadei, Please double-check the tests. Some of them fail. Also, please move the `experiments` under the `functional_tests` folder.

**Pull request merged**